submissions/7SergSmith7/Friends-app/app.js

```diff
@@ -0,0 +1,160 @@
+  const { gender, name, location, dob, phone, picture } = contact;
+  return `<div class="contact__item" data-name="${name.first}">
+            <img class="contact__image" src=${picture.large} alt=${name.first}>
+            <h3 class="contact__title">${name.first} ${name.last}</h3>
+            <p class="contact__phone">${phone}</p>   
+            <div class="contact__additional">
+              <div class="contact__additional__gender">
+                <h4 class="additional__gender__title">sex</h4>
+                <p class="additional__gender_item">${gender}</p>
+              </div>
+              <div class="contact__additional__age">
+                <h4 class="additional__age__title">age</h4>
+                <p class="additional__age_item">${dob.age}</p>
+              </div>
+              <div class="contact__additional__location">
+              <h4 class="additional__location__title">location</h4>   
+              <p class="additional__location_item">${location.country}</p>     
+              </div>
+            </div>
+          </div>`;
+}
+
+optionsMenu.addEventListener("click", onOptionsClick);
+searchInput.addEventListener("change", onChangeSearch);
+
+function onChangeSearch() {
+  contacts = contacts.filter((element) => {
+    return element.name.first
+      .toLowerCase()
+      .includes(searchInput.value.toLowerCase());
+  });
+  renderContacts(contacts);
+}
+
+function renderContacts(contacts) {
+  contactsField.innerHTML = contacts
+    .map((contact) => renderContactTemplate(contact))
+    .join("");
+}
+
+function onOptionsClick(e) {
+  const selectedOption = e.target.classList.value;
+
+  switch (selectedOption) {
+    case "name-sort__z-a":
+      sortName(contacts);
+      renderContacts(sortWithFilter(contacts, filter));
+
+      break;
+    case "name-sort__a-z":
+      sortName(contacts);
+      contacts = contacts.reverse();
+      renderContacts(sortWithFilter(contacts, filter));
+
+      break;
+    case "age-sort__more":
+      sortAge(contacts);
+
+      renderContacts(sortWithFilter(contacts, filter));
+
+      break;
+    case "age-sort__less":
+      sortAge(contacts);
+      contacts = contacts.reverse();
+      renderContacts(sortWithFilter(contacts, filter));
+
+      break;
+    case "gender-filter__man":
+      filter = "male";
+      renderContacts(filterGender(contacts, filter));
+
+      break;
+    case "gender-filter__woman":
+      filter = "female";
+      renderContacts(filterGender(contacts, filter));
+      break;
+    case "options__blocks__reset-btn":
+      resetOptions();
+      break;
+    // case "options__search__icon":
+    //   renderContacts(searchName(searchInput));
+    //   break;
+    default:
+      break;
+  }
+}
+
+function resetOptions() {
+  copyData();
+  filter = "";
+  searchInput.value = "";
+  console.log(contacts);
+  renderContacts(contacts);
+}
+
+function sortAge(contact) {
+  contacts = contact.sort((a, b) => a.dob.age - b.dob.age);
```

Comment:
Why are you assigning result of `sort`? `Array#sort` sorts array in place

Reply:
Didn't understand a little what needs to be done(

Reply:
Answer question :) Sort sorts array "in place". It will mutate your array in other words.

Reply:
Just 
{contact.sort((a, b) => a.dob.age - b.dob.age);}
 ?

Reply:
Kind of. But I still don't understand why you needed return :)